### PR TITLE
Add IntentClassifierConfiguration (with interfaces)

### PR DESCRIPTION
**Pull request by andrii-movadex**

### CODE REVIEW

File: src/LanguageProcessorEngine/Configuration/BaseLanguageProcessorConfiguration.php [hunk 1]
```diff
@@ -7,11 +7,30 @@
...
         return new ComponentConfigurationKeyWithName($this->configurationName);
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function setType(string $type): void
+    {
+        $this->type = $type;
+    }
```
Comment (elliotmassen) [inline src/LanguageProcessorEngine/Configuration/BaseLanguageProcessorConfiguration.php:26]: Let's remove this setter method. Please see `InterpreterWithEntitiesConfiguration` for an example of the pattern we use for setting properties in configurations. We will expect the `type` data to be present in the `$data` parameter of the constructor of this configuration (stored in the `configuration` field on the `ComponentConfiguration` model).

File: src/LanguageProcessorEngine/Tests/Configuration/ConfigurationTest.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace OpenDialogAi\LanguageProcessorEngine\Tests\Configuration;
+
+use OpenDialogAi\Core\Components\Configuration\ComponentConfigurationKeyWithName;
+use OpenDialogAi\Core\Tests\TestCase;
+use OpenDialogAi\LanguageProcessorEngine\Configuration\BaseLanguageProcessorConfiguration;
+
+class ConfigurationTest extends TestCase
```
Comment (elliotmassen) [inline src/LanguageProcessorEngine/Tests/Configuration/ConfigurationTest.php:9]: Let's add a test here to check that the `type` property is successfully get/set.

**Review by elliotmassen — CHANGES_REQUESTED** [review]

File: src/LanguageProcessorEngine/Configuration/BaseLanguageProcessorConfiguration.php [hunk 3]
```diff
@@ -7,11 +7,30 @@
 
 class BaseLanguageProcessorConfiguration extends BaseConfiguration implements LanguageProcessorConfigurationInterface
 {
+    const TYPE_KEY = 'type';
```
Comment (elliotmassen) [inline src/LanguageProcessorEngine/Configuration/BaseLanguageProcessorConfiguration.php:10]: Let's just call this TYPE for simplicity.

```suggestion
    const TYPE = 'type';
```

**Review by elliotmassen — CHANGES_REQUESTED** [review]
Thanks Andrii! Just two final small tweaks to make.

File: src/LanguageProcessorEngine/Configuration/BaseLanguageProcessorConfiguration.php [hunk 4]
```diff
@@ -7,14 +7,15 @@
 
 class BaseLanguageProcessorConfiguration extends BaseConfiguration implements LanguageProcessorConfigurationInterface
 {
-    const TYPE_KEY = 'type';
+    const TYPE = 'type';
     private string $type;
+    protected static array $requiredAttributes = [self::TYPE];
```
Comment (elliotmassen) [inline src/LanguageProcessorEngine/Configuration/BaseLanguageProcessorConfiguration.php:12]: Apologies this is my bad for not being specific enough, this wants to be the `$required` property from `BaseConfiguration`.

```suggestion
    protected static array $required = [self::TYPE];
```

File: src/LanguageProcessorEngine/Tests/Configuration/ConfigurationTest.php [hunk 5]
```diff
@@ -43,9 +44,15 @@ public function testSetupOfConfigurationTypeField()
...
-        $configuration = BaseLanguageProcessorConfiguration::create('my_config', [BaseLanguageProcessorConfiguration::TYPE_KEY => $configurationType]);
+        $configuration = BaseLanguageProcessorConfiguration::create('my_config', [BaseLanguageProcessorConfiguration::TYPE => $configurationType]);
 
         // Assert
         $this->assertEquals($configurationType, $configuration->getType());
     }
+
+    public function testConfigurationCreationWithoutTypeField()
+    {
+        $this->expectException(ErrorException::class);
```
Comment (elliotmassen) [inline src/LanguageProcessorEngine/Tests/Configuration/ConfigurationTest.php:55]: I think we'd expect `InvalidConfigurationDataException` here instead.

**Review by elliotmassen — CHANGES_REQUESTED** [review]

File: src/LanguageProcessorEngine/Configuration/BaseLanguageProcessorConfiguration.php (hunk 4 above)
Comment (andrii-movadex) [reply src/LanguageProcessorEngine/Configuration/BaseLanguageProcessorConfiguration.php]: No worries, I also could be more attentive. Fixed.

**Review by elliotmassen — APPROVED** [review]

**Pull request merged**